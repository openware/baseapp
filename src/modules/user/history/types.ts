import { CommonState } from '../../types';

export interface PublicTrade {
    id: number;
    price: string;
    total: string;
    amount: string;
    market: string;
    created_at: string;
    taker_type: string;
}

export interface PrivateTrade extends PublicTrade {
    side?: string;
    order_id?: number;
}

export interface PrivateTradeEvent {
    id: number;
    price: string;
    total: string;
    amount: string;
    market: string;
    created_at: string;
    taker_type: string;
    side?: string;
    order_id?: number;
}

export interface PrivateTradesState extends CommonState {
    list: PrivateTrade[];
}

export type MakerType = 'buy' | 'sell';

export interface Withdraw {
    currency: string;
    id: number;
    type: string;
    amount: string;
    fee: string;
    blockchain_txid: string;
    rid: string;
    state: string;
    created_at: string;
    updated_at: string;
    completed_at: string;
    done_at: string;
}

export interface Deposit {
    currency: string;
    id: number;
    amount: string;
    fee: string;
    txid: string;
    created_at: string;
    confirmations: number;
    completed_at: string;
    state: string;
}

export type WalletHistoryElement = Withdraw | Deposit | PrivateTrade;
export type WalletHistoryList = WalletHistoryElement[];
