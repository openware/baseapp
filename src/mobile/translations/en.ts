export const en = {
    'page.mobile.header.signIn': 'Log In',
    'page.mobile.footer.home': 'Home',
    'page.mobile.footer.orders': 'Orders',
    'page.mobile.footer.trading': 'Trade',
    'page.mobile.footer.wallets': 'Wallets',
    'page.mobile.footer.profile': 'Profile',

    'page.mobile.tradingTabs.label.createOrder': 'Create Order',
    'page.mobile.tradingTabs.label.charts': 'Charts',
    'page.mobile.tradingTabs.label.orders': 'Orders',

    'page.mobile.charts.label.orderBook': 'Order book',
    'page.mobile.charts.label.trades': 'Trades',

    'page.mobile.orderButtons.buy': 'Buy {base_unit}',
    'page.mobile.orderButtons.sell': 'Sell {base_unit}',

    'page.mobile.orders.tabs.open': 'Open',
    'page.mobile.orders.tabs.all': 'All',
    'page.mobile.orders.cancelAll': 'Cancel All',

    'page.mobile.orders.open.title': 'Open Orders',
    'page.mobile.orders.header.orderType.buy.market': 'Buy market',
    'page.mobile.orders.header.orderType.buy.limit': 'Buy limit',
    'page.mobile.orders.header.orderType.sell.market': 'Sell market',
    'page.mobile.orders.header.orderType.sell.limit': 'Sell limit',
    'page.mobile.orders.header.amount': 'Amount',
    'page.mobile.orders.header.price': 'Price',
    'page.mobile.orders.header.filled': 'Filled',
    'page.mobile.orders.header.cancel': 'Cancel',

    'page.mobile.currentMarketInfo.search.placeholder': 'Search',
    'page.mobile.currentMarketInfo.volume': 'Vol',
    'page.mobile.currentMarketInfo.high': 'High',
    'page.mobile.currentMarketInfo.low': 'Low',

    'page.header.signIn.forgotPassword': 'Forgot password?',
    'page.header.signIN.noAccountYet': 'No account yet?',
    'page.header.signUp.terms': 'I  agree all statements in terms of service',
    'page.header.signUp.alreadyRegistered': 'Already registered?',

    'page.mobile.reset.header': 'Resend confirmation',
    'page.mobile.signin.kyc.header': '2FA Verification',

    'page.mobile.wallets.deposit.history': 'Deposit history',
    'page.mobile.wallets.withdraw.history': 'Withdraw history',

    'page.mobile.copy.text': 'Copy Address',
    'page.mobile.wallet.deposit.generate': 'Generate deposit address',
    'page.mobile.wallet.deposit.generating': 'Generating address...',
};
