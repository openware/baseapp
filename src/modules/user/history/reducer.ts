import { getUnique } from '../../../helpers/getUnique';
import { HistoryActions } from './actions';
import {
    HISTORY_DATA,
    HISTORY_ERROR,
    HISTORY_FETCH,
    HISTORY_PUSH_FINISH,
    HISTORY_RESET,
} from './constants';
import { WalletHistoryList } from './types';

export interface HistoryState {
    list: WalletHistoryList;
    fetching: boolean;
    total: number;
    page: number;
}

const initialState: HistoryState = {
    list: [],
    fetching: false,
    total: 0,
    page: 0,
};


export const historyReducer = (state = initialState, action: HistoryActions) => {
    switch (action.type) {
        case HISTORY_FETCH:
            return { ...state, fetching: true };
        case HISTORY_DATA:
            return {
                ...state,
                list: action.payload.list,
                fetching: false,
                page: action.payload.page,
                total: action.payload.total,
            };
        case HISTORY_ERROR: {
            return { ...state, list: [], fetching: false, total: 0, page: 0 };
        }
        case HISTORY_RESET: {
            return { ...state, list: [], total: 0, page: 0 };
        }
        case HISTORY_PUSH_FINISH: {
            let list = [...action.payload];
            list = getUnique(list, 'id');
            return { ...state, list: list };
        }
        default:
            return state;
    }
};
