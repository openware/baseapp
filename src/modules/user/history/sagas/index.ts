import { takeLatest } from 'redux-saga/effects';
import { HISTORY_FETCH, HISTORY_PUSH_EMIT } from '../constants';
import { historyPushSaga } from './historyPushSaga';
import { historySaga } from './historySaga';

export function* rootHistorySaga() {
    yield takeLatest(HISTORY_PUSH_EMIT, historyPushSaga);
    yield takeLatest(HISTORY_FETCH, historySaga);
}
