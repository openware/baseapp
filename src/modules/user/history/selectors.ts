import { RootState } from '../../index';
import { WalletHistoryList } from './types';

export const selectHistory = (state: RootState): WalletHistoryList =>
    state.user.history.list;

export const selectHistoryTotal = (state: RootState): number =>
    state.user.history.total;

export const selectCurrentPage = (state: RootState): number =>
    state.user.history.page;

export const selectPageCount = (state: RootState, limit): number =>
    Math.ceil(state.user.history.total / limit);

export const selectFirstElemIndex = (state: RootState, limit): number =>
    (state.user.history.page * limit) + 1;

export const selectLastElemIndex = (state: RootState, limit: number): number => {
    if (state.user.history.total >= 0) {
        if ((state.user.history.page * limit) + limit > selectHistoryTotal(state)) {
            return selectHistoryTotal(state);
        } else {
            return (state.user.history.page * limit) + limit;
        }
    } else {
        return (state.user.history.page * limit) + state.user.history.list.length;
    }
};

export const selectNextPageExists = (state: RootState, limit: number): boolean => {
    if (state.user.history.total >= 0) {
        return (state.user.history.page + 1) < selectPageCount(state, limit);
    } else {
        return state.user.history.list.length === limit;
    }
};

export const selectHistoryLoading = (state: RootState): boolean =>
    state.user.history.fetching;
