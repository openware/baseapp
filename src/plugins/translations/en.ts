export const en = {
    'page.body.ieo.profile.info.price.close.in': 'Presale Close In',
    'page.body.ieo.profile.info.price': 'Price',
    'page.body.ieo.profile.info.goal': 'Goal',
    'page.body.ieo.profile.info.progress': 'Progress',
    'page.body.ieo.profile.info.description': 'Web Monetization with Decentralized Blockchains, Masternodes, Merged Mining and Interoperable Smart-contracts web Monetization with Decentralized Blockchains, Masternodes, Merged Mining and Interoperable Smart-contracts web Monetization with Decentralized Blockchains, Masternodes, Merged Mining and Interoperable Smart-contracts web Monetization with Decentralized Blockchains, Masternodes, Merged Mining and Interoperable Smart-contracts web Monetization with Decentralized Blockchains, Masternodes, Merged Mining and Interoperable Smart-contracts',

    'page.body.ieo.profile.details.header': 'Details',
    'page.body.ieo.profile.details.name': 'Name',
    'page.body.ieo.profile.details.full.name': 'Full name',
    'page.body.ieo.profile.details.session.supply': 'Session supply',
    'page.body.ieo.profile.details.total.supply': 'Total supply',
    'page.body.ieo.profile.details..ieo.ratio': 'IEO Ratio',
    'page.body.ieo.profile.details.pre.sale.price': 'Pre-sale price',
    'page.body.ieo.profile.details.minimum.contribution': 'Minimum contribution',
    'page.body.ieo.profile.details.ieo.start': 'IEO Start Time',
    'page.body.ieo.profile.details.ieo.end': 'IEO End Time',

    'page.body.ieo.profile.details.technological.foundation': 'Technological foundation',
    'page.body.ieo.profile.details.website': 'Website',
    'page.body.ieo.profile.details.whitepaper': 'Whitepaper',
    'page.body.ieo.profile.details.telegram': 'Telegram',
    'page.body.ieo.profile.details.twitter': 'Twitter',
    'page.body.ieo.profile.details.bicointalk': 'BikoinTalk',

    'page.body.ieo.profile.introduction.header': 'Project Introduction',
    'page.body.ieo.profile.introduction.welcome': 'Welcome to the new investment horizons.',
    'page.page.body.ieo.profile.introduction.coin.market': 'The Open Ware coin market,and EOS-based platform to buy and sell natural and lab-grown diamonds, heirloom or other jewellery, is opening the formerly closed diamond market for everybody.',
    'page.page.body.ieo.profile.introduction.owc.present': 'OWC willpresent a marketplace and a trading platform for businesses and individuals working as a reliable tool for B2B, B2C or P2P relations.',
    'page.page.body.ieo.profile.introduction.buy.directly': 'Buy directly from diamond producers, wholesalers and individual market participants. Sell your diamond ring or other jewellery to anyone in the world. Open your own shop with OWC token-powered services, launch ads and promotional campaigns.',
    'page.page.body.ieo.profile.introduction.other.options': 'Other options available for OWC token are:',
    'page.page.body.ieo.profile.introduction.option.first': '- transaction insurance',
    'page.page.body.ieo.profile.introduction.option.second': '- storageservices',
    'page.page.body.ieo.profile.introduction.option.third': '- relevant search, helping to find the best offers based on a particular request',
    'page.page.body.ieo.profile.introduction.option.fourth': '- trade statistics, giving a deeper glimpse on the diamond marketsâ€™ supply and demand.',
    'page.page.body.ieo.profile.introduction.complex.deal': 'For complex deals our consultants are ready to help and manage the technical aspect, assuring the deal will be done the fastest way possible. For clients who wish to learn about the special offers as soon as they appear on the platform, OWC will offer premium support by personal manager.',
    'page.page.body.ieo.profile.introduction.togetether.funcrtions': 'All together these functions will serve to bring the best trading experience and open a new world of possibilities for the open diamond market participants.',
    'page.page.body.ieo.profile.introduction.market.remains': 'The OWC market remains a closed area where most investors are not allowed. We believe that access to the OWC market is a natural right of any individual and build Openware Market to operate worldwide.',
    'page.page.body.ieo.profile.introduction.dom': 'The DOM platform will allow users to choose between natural or lab-grown diamonds from the manufacturers or individuals and individually agree on the prices.',
    'page.page.body.ieo.profile.introduction.dom.willoffer': 'DOM willoffer variety of payment methods, including banking, cash or cryptocurrency transactions. OWC will be built on EOS blockchain ecosystem and operate without any fees for both counterparties.',
    'page.page.body.ieo.profile.introduction.businesscompanies': 'Businesscompanies will be able to open virtual branches on OWC, using value-added services of the platform. DOM will also introduce a secured smart contract system, allowing to implement any business logic and solve the safety issues.',
    'page.page.body.ieo.profile.introduction.learn.more': 'Learn more on the company\'s website www.openware.com',

    'page.body.ieo.card.start.date': 'Start date',
    'page.body.ieo.card.end.date': 'End date',
    'page.body.ieo.card.countdown': 'Countdown',
    'page.body.ieo.card.raised': 'Raised',
    'page.body.ieo.card.supply': 'Supply',
    'page.body.ieo.card.end': 'End',

    'page.body.ieo.details.order.youWillGet': 'You will get',
    'page.body.ieo.details.order.youWillPay': 'You will pay',
    'page.body.ieo.details.order.content.blur.title': 'Sales will be start after',

};
