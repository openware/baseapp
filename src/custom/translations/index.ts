import { en } from '../../translations/en';
import { ru } from './ru';

// tslint:disable:no-submodule-imports
import localeRu from 'react-intl/locale-data/ru';
// tslint:enable

export const customLocaleData = ([...localeRu]);

export type LangType = typeof en;

export const customLanguageMap = {
    ru,
};
