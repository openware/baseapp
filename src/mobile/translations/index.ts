import { en } from './en';

export {
    en,
};
