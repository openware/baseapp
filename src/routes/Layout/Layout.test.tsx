import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import * as messages from '../../translations/en';
import { Layout, LayoutProps } from './';

jest.mock('react-ga');

const defaults: LayoutProps = {
    colorTheme: 'basic',
    currentMarket: undefined,
    user: {
        email: '',
        level: 0,
        otp: false,
        role: '',
        state: '',
        uid: '',
    },
    isLoggedIn: false,
    logout: jest.fn(),
    userFetch: jest.fn(),
    walletsReset: jest.fn(),
    history: createBrowserHistory(),
};

// tslint:disable-next-line: no-shadowed-variable
const nodeWithIntlProp = ((node, { intl }) => {
    return React.cloneElement(node, { intl });
});

const intlProvider = new IntlProvider({ locale: 'en', messages }, {});
const { intl } = intlProvider.getChildContext();

const setup = (props: Partial<LayoutProps> = {}) =>
    shallow(nodeWithIntlProp(<Layout {...{ ...defaults, ...props }} />, { intl }), {
    context: { intl },
});

describe.skip('Layout component', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });
});
