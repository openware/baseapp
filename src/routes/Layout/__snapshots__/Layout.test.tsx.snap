// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render 1`] = `ShallowWrapper {}`;
